### ISSUE TITLE
Support overflow-safe casts

### ISSUE THREAD

**Issue by tdz** [picotm/picotm#131]
A new module for overflow casts might be added. Casting between types might overflow/underflow the resulting type. The module's cast operation shall test this and report an overflow/underflow error is the result is out of range of the result type. Something like this:

~~~ c
picotm_begin
  int ival = 12345;
  char cval = convertc_tx(ival);
picotm_commit
  // handle error here
picotm_end
~~~

The ```convertc_tx()``` function converts an integer value to the specified value of type ```char``` and generates the error if necessary.

This requires new generic error codes and a new module in the modules/ directory. It's a smaller, self-contained change, so it's labeled as good first bug.

**Comment by tdz** [comment]
Or maybe call the function `cast_c_tx()`, `cast_ul_tx()`, etc.

**Comment by tdz** [comment]
Fixed by PR #244
